src/state/map/map.state.listeners.ts

```diff
@@ -76,7 +76,7 @@ export class MapSateListener {
   ): Layer[] {
     if (contextHasChanged(newContext, oldContext)) {
       return (
-        newContext?.layers?.reduce(
+        newContext.layers?.reduce(
```

Comment:
`layers` also can't be falsy

Reply:
you can fore typing with `as` 
`const layer: Layer | undefined`
`(layer as Layer)` can't be undefined.